### PR TITLE
Remove RecoVertex/NuclearInteractionProducer package

### PR DESCRIPTION
**Pull request by guitargeek**
#### PR description:

I saw @perrotta stumbled recently across a clang issue in the __RecoVertex/NuclearInteractionProducer__ package https://github.com/cms-sw/cmssw/pull/27174, and he noted that the class that he was fixing might have a bug and that it's not used in production anywhere. Since the package was originally intended to be used in PF, he suggested to take a look at it in the scope of the recent PF cleanup.

So here we go: according to indico [1] the package was part of a project to improve tracking in PF by finding the vertices of nuclear interactions and refitting the tracks. However, it seems abandoned as there were no non-technical changes since 2008 [2] and the person responsible for that project (Roberfroid Vincent) apparently left CMS in 2008.

The 3 config files in RecoParticleFlow that were referring to NuclearInteractionProducer are also orphans, so they can be safely removed.

In conclusion, this package is indeed a good candidate for the cleanup.

@hatakeyamak @bendavid

[1] https://indico.cern.ch/event/32195/contributions/746713/attachments/622247/856187/vincent.pdf#search=nuclearinteractionproducer
[2] https://github.com/cms-sw/cmssw/commits/master/RecoVertex/NuclearInteractionProducer

#### PR validation:

CMSSW compiles and local matrix tests pass.

#### if this PR is a backport please specify the original PR:

No backport intended.

### CODE REVIEW

**Comment by slava77** [comment]
@cmsbuild please test

**Comment by perrotta** [comment]
+1
- The whole RecoVertex/NuclearInteractionProducer packege and related PF configs are clearly not used since a while, and possibly also bugged somewhere
- If @hatakeyamak @bendavid have nothing in contrary, this cleanup is welcome
- Jenkins tests pass

**Comment by hatakeyamak** [comment]
I don't see any harm of this cleaning, but can you give me 1-2 days to digest what this is about and just in case I am thinking of letting people know about this and some of recent cleanups via HN.

**Comment by Dr15Jones** [comment]
+1

**Comment by hatakeyamak** [comment]
I think it's ok to go ahead and do this cleanup.

**Comment by fabiocos** [comment]
+1
